File: One Task/Task 1/c programming/37.c
// Binary Search
#include <stdio.h>

void binarySearch(int arr[], int l, int r, int x, int n);

int main() {
    int arr[] = {12, 23, 45, 67, 78};

    binarySearch(arr, 0, 4, 67, 5);

    return 0;
}

void binarySearch(int arr[], int l, int r, int x, int n) {
    int mid = (l + r)/2;
    if(l <= r) {
        if(arr[mid] == x) {
            printf("%d found at position %d", x, mid+1);
        }else if(x > arr[mid]) {
            binarySearch(arr, mid + 1, r, x, n-1);
        }else if(x < arr[mid]) {
            binarySearch(arr, l, mid - 1, x, n-1);
        }
    }else {
        printf("%d found at position %d", x, mid+1);
    }
}

File: Without Node Module/Xampp/htdocs/core/week-1/c-programs/programs/37.php
<?php
    $array = [1, 2, 3, 5, 8];
    $searchElement = 5;

    function binarySearch($array, $searchElement) {
        $low = 0;
        $high = count($array) - 1;

        while ($low <= $high) {
            $mid = floor(($low + $high) / 2);

            if ($array[$mid] == $searchElement) {
                return $mid;
            } elseif ($array[$mid] < $searchElement) {
                $low = $mid + 1;
            } else {
                $high = $mid - 1;
            }
        }

        return -1;
    }

    $result = binarySearch($array, $searchElement);
    echo "Binary Search: Element $searchElement found at index $result";
?>
